src/main/java/com/coursesPlatform/coursePortfolio/OfferService.java

```diff
@@ -10,28 +14,95 @@
+    }
+
+    List<OfferDTO> findAllOffers() {
+        return offerRepository.findAll()
+                .stream()
+                .map(this::convertOfferToDTO)
+                .collect(toList());
+    }
+
+    List<OfferDTO> findAllOffersOfCustomer(String mail) {
+        return offerRepository.findByCustomerMail(mail)
+                .stream()
+                .map(this::convertOfferToDTO)
+                .collect(toList());
+
+    }
+
+    @Transactional
+    OfferDTO create(OfferDTO offerDTO) {
+        Offer offer = offerRepository.save(convertDTOToOffer(offerDTO));
+        List<Lecture> lectures = offer.getLectures();
+        BigDecimal price = sumPriceOfLectures(lectures);
+        int duration = sumDurationOfLectures(lectures);
+        offer.setSummaryPrice(price);
+        offer.setSummaryDuration(duration);
+        return convertOfferToDTO(offer);
+    }
+
+    private BigDecimal sumPriceOfLectures(List<Lecture> lectures) {
         return lectures.stream()
-                .map(LectureDTO::getPrice)
+                .map(Lecture::getPrice)
                 .reduce(BigDecimal.ZERO, BigDecimal::add);
     }
 
-    int sumDurationOfLectures(List<LectureDTO> lectures) {
+    private int sumDurationOfLectures(List<Lecture> lectures) {
         return lectures.stream()
-                .mapToInt(LectureDTO::getDuration)
+                .mapToInt(Lecture::getDuration)
                 .sum();
     }
 
     private OfferDTO convertOfferToDTO(Offer offer) {
+        Customer customer = offer.getCustomer();
         Category category = offer.getCategory();
         Course course = offer.getCourse();
         List<Lecture> lectures = offer.getLectures();
         return new OfferDTO(
+                customer.getMail(),
                 category.getTitle(),
                 course.getTitle(),
                 lectures.stream().map(Lecture::getTitle).collect(toList()),
                 offer.getSummaryPrice(),
                 offer.getSummaryDuration()
         );
     }
+
+    private Offer convertDTOToOffer(OfferDTO offerDTO) {
+        Customer customer = customerRepository.findCustomerByMail(offerDTO.getMail()).orElseThrow(CustomerNotFoundException::new);
+        Category category = categoryRepository.findCategoryByTitle(offerDTO.getCategoryTitle()).orElseThrow(CategoryNotFoundException::new);
+        Course course = courseRepository.findCourseByTitle(offerDTO.getCourseTitle()).orElseThrow(CourseNotFoundException::new);
+        List<Lecture> lectures = checkIfLectureListIsNotEmpty(offerDTO.getLecturesTitle());
+        BigDecimal summaryPrice = offerDTO.getSummaryPrice();
+        int summaryDuration = offerDTO.getSummaryDuration();
+        return new Offer(
+                customer,
+                category,
+                course,
+                lectures,
+                summaryPrice,
+                summaryDuration
+        );
+    }
+
+    private List<Lecture> checkIfLectureListIsNotEmpty(List<String> lectures) {
```

Comment:
Method name missleading. It suggest validation while purpose of method is to find all Lectures with given mails.